### PR TITLE
ADAPT-2437: Create embedCard

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- Create embedCard
*Unless additional styling is needed, I believe this is ready for review 😄 I'll update Storyblok to include additional variations of embeds on Monday.

# Review By (Date)
- tbd

# Review Tasks
## Setup tasks and/or behavior to test
1. Check out this branch
2. Run local dev or view [Netlify Preview](https://deploy-preview-35--stanford-alumni.netlify.app/test/embed-card-test/)

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Associated Issues and/or People
- [ADAPT-2437](https://stanfordits.atlassian.net/browse/ADAPT-2437)

### CODE REVIEW

**Comment by rebeccahongsf** [comment]
Additional styling is needed but I'd love feedback on the functionality when you have the chance @sherakama @yvonnetangsu 😸

**Comment by yvonnetangsu** [comment]
> Additional styling is needed but I'd love feedback on the functionality when you have the chance @sherakama @yvonnetangsu 😸

I will review but go enjoy your long weekend 😸

File: gatsby-node.js [hunk 1]
```diff
@@ -113,12 +121,24 @@ exports.onCreateWebpackConfig = ({
...
+      module: {
+        rules: [
+          {
+            test: /postscribe/,
+            use: loaders.null(),
+          },
+        ],
+      },
+    });
+  }
```
Comment (sherakama) [inline gatsby-node.js:143]: Oh interesting, so this is essentially telling webpack to skip processing postscribe and leave it "as is"?

File: src/components/cards/embedCard.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      // Create a 'tiny' document and parse the html string.
+      // https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment
+      const miniDom = document.createRange().createContextualFragment(html);
+      // Clear the container.
+      myEmbed.current.innerHTML = "";
+      // Append the new content.
+      myEmbed.current.appendChild(miniDom);
+    }
+  }, []);
+
```
Comment (sherakama) [inline src/components/cards/embedCard.js:45]:
```suggestion
  if (!html) { 
    return null ;
  }
```

File: src/components/cards/embedCard.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  return (
+    <SbEditable content={blok}>
+      {html.includes("script") ? (
+        <>
+          <div id={uniqueId} />
```
Comment (sherakama) [inline src/components/cards/embedCard.js:50]:
```suggestion
          <div id={uniqueId} aria-live="polite" aria-busy="{!scriptLoaded}" />
```

File: src/components/cards/embedCard.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          {!scriptLoaded && (
+            <>
+              <Loader
+                type="TailSpin"
+                color="#00BFFF"
+                height={50}
+                width={50}
+                timeout={30000}
+                aria-busy="true"
```
Comment (sherakama) [inline src/components/cards/embedCard.js:59]:
```suggestion
```

**Review by sherakama — CHANGES_REQUESTED** [review]
A couple of minor changes but otherwise this is working great. Nice work on this.

File: gatsby-node.js (hunk 1 above)
Comment (rebeccahongsf) [reply gatsby-node.js:143]: Yes! 'Tis the gatsby suggested way of excluding certain packages from SSR 😄

Comment (sherakama) [reply gatsby-node.js:143]: Very cool. Do you have a link to that documentation?

**Review by sherakama — APPROVED** [review]
GTG by me.

File: gatsby-node.js (hunk 1 above)
Comment (rebeccahongsf) [reply gatsby-node.js:143]: @sherakama [Gatsby Debugging documentation: How to check if window is defined](https://www.gatsbyjs.com/docs/debugging-html-builds/#how-to-check-if-window-is-defined)😃

**Comment by yvonnetangsu** [comment]
Yay - yeah no particular styling needed for this. Thanks for your work, @rebeccahongsf and thanks for reviewing @sherakama  🥳

**Pull request merged**